### PR TITLE
Feature/sql chain

### PR DESCRIPTION
**Pull request by zeqing-guo**

### CODE REVIEW

File: merkle/trie.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+package merkle
+
+import (
+	"errors"
+
+	"github.com/tchap/go-patricia/patricia"
+	"github.com/thunderdb/ThunderDB/crypto/hash"
+)
+
+type Trie struct {
```
Comment (covenantdb) [inline merkle/trie.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported type Trie should have comment or be unexported

File: merkle/trie.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+// Insert serializes key into binary and computes its hash,
+// then stores the (hash(key), value) into the trie
+func (trie *Trie) Insert(key []byte, value []byte) (inserted bool) {
+	hashedKey := hash.HashB(key)
+
+	inserted = trie.trie.Insert(hashedKey, value)
+	return
+}
+
+func (trie *Trie) Get(key []byte) ([]byte, error) {
```
Comment (covenantdb) [inline merkle/trie.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method Trie.Get should have comment or be unexported

**Review by auxten — COMMENTED** [review]
Keep License Apache2.0

**Review by auxten — APPROVED** [review]
so far so good

**Pull request merged**